lib/lightstep/instrumentation.rb

```diff
@@ -0,0 +1,41 @@
+module LightStep
+  # Instrument an operation. This is may be called by any LightStep component to
+  # record an instrumentation event. This is a compatible API to
+  # ActiveSupport::Notifications.
+  #
+  # event   - Name of the event as a symbol. The event is published under the
+  #           lightstep namespace automatically.
+  # payload - Hash payload for the event. Available keys are based on the event
+  #           being published.
+  #
+  # Returns the result of executing the block.
+  def self.instrument(event, payload = {}, &block)
+    if @instrumenter
+      block ||= proc {}
+      @instrumenter.instrument("#{event}.lightstep", payload, &block)
+    elsif block
+      block.call(payload)
```

Comment:
Will behavior be different if `@instrumenter` is present or not? It seems that when instrumenter is true, `block` is called with no parameters, and when it's not, block is called with the payload as a parameter. Reading other ActiveSupport::Notification examples, I don't often see the payload being an argument to the block being instrumented.